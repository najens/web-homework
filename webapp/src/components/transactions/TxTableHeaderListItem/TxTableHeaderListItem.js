import React from 'react'
import { css } from '@emotion/core'
import { ArrowDropUp, ArrowDropDown } from '@material-ui/icons'
import { string, func } from 'prop-types'

export default function TxTableHeaderListItem ({ asc, desc, heading, orderBy, refetchTransactions, setOrderBy, width }) {
  const isAsc = orderBy === asc
  const isDesc = orderBy === desc
  const isFiller = heading === ''

  function handleHeaderClick () {
    const newOrderBy = isDesc ? asc : desc
    refetchTransactions({ orderBy: newOrderBy })
    setOrderBy(newOrderBy)
  }

  return (
    <th css={headerStyle} style={{ width }}>
      {!isFiller && (
        <button css={buttonStyle} onClick={handleHeaderClick}>
          <span>{heading}</span>
          {isAsc && <ArrowDropUp style={{ fontSize: 24 }} />}
          {isDesc && <ArrowDropDown style={{ fontSize: 24 }} />}
        </button>
      )}
    </th>
  )
}

TxTableHeaderListItem.propTypes = {
  asc: string,
  desc: string,
  heading: string,
  orderBy: string,
  refetchTransactions: func,
  setOrderBy: func,
  width: string
}

const headerStyle = css`
  border: 1px solid #b4c1ec;
`

const buttonStyle = css`
  align-items: center;
  background: none;
  border: none;
  display: flex;
  flex-direction: row;
  height: 24px;
  justify-content: flex-start;
  padding: 0px 6px;
  width: 100%;
  -moz-appearance: none;
  -webkit-appearance: none;

  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
  }
`
