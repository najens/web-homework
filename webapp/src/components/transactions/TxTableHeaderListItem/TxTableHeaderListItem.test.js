import React from 'react'
import { create } from 'react-test-renderer'
import TxTableHeaderListItem from './TxTableHeaderListItem'

// Only testing render since the business logic tested in parent component
describe('TxTableHeaderListItem component', () => {
  it('renders correctly', () => {
    const testRenderer = create(
      <TxTableHeaderListItem
        asc='INSERTED_AT_ASC'
        desc='INSERTED_AT_DESC'
        heading='Date'
        key='date'
        orderBy='INSERTED_AT_DESC'
        refetchTransactions={() => null}
        setOrderBy={() => null}
        width='10%'
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
