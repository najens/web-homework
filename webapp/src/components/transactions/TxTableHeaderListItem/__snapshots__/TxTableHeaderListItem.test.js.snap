// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTableHeaderListItem component renders correctly 1`] = `
.emotion-1 {
  border: 1px solid #b4c1ec;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 24px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0px 6px;
  width: 100%;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  cursor: pointer;
}

<th
  className="emotion-1"
  style={
    Object {
      "width": "10%",
    }
  }
>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    <span>
      Date
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "fontSize": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </button>
</th>
`;
